receipt_handler.go

```diff
@@ -0,0 +1,89 @@
+package node
+
+import (
+	"log"
+	"sync"
+)
+
+// Takes care of maintaining maps and insures that we know which interfaces are reachable where.
+type ReceiptHandler interface {
+	AddConnection(NodeAddress, ReceiptConnection)
+	AddSentPacket(p Packet, src, next NodeAddress)
+	SendReceipt(PacketReceipt)
+}
+
+type packetRecord struct {
+	destination NodeAddress
+	src         NodeAddress
+	next        NodeAddress
+	hash        PacketHash
+}
+
+type ReceiptAction interface {
+	Receipt(PacketHash)
+}
+
+type receiptImpl struct {
```

Comment:
I've seen you use this style of naming a few times now, what does Impl mean stand for?